src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -71,9 +71,12 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
 
     DockerBuildImage() {
         ext.getImageId = { imageId }
-
+        onlyIf {
+            boolean runTask = previouslyBuiltImageExists() ? false : true
+            if (!runTask) { logger.quiet("${this.name}: UP-TO-DATE") }
```

Comment:
@bmuschko it's a hack no doubt about it. Trying to use the outputs.upToDateWhen {} does not actually work inside of a custom task. Searching the internets I found others with the same problem and it's supposedly a known gradle issue. The workaround suggested was to use onlyIf {} as that works as expected.